src/index.ts

```diff
@@ -20,72 +20,135 @@
+  promptStaticClientInfo,
+} from "./prompts";
 
-import express from "express";
-const app = express();
-const PORT = 3001;
-const iriBase = `http://localhost:${PORT}`;
-const storage = new InMemoryStorage();
+type InputOptions = {
+  solidIdentityProvider?: string;
+  clientName?: string;
+  registrationType?: "static" | "dynamic";
+};
+
+type ValidatedOptions = {
+  solidIdentityProvider: string;
+  registrationType: "static" | "dynamic";
+  clientName?: string;
+};
 
-// Initialised when the server comes up and is running...
-let session: Session;
+async function main(): Promise<void> {
+  // Get CLI arguments
+  const argv = require("yargs/yargs")(process.argv.slice(2))
+    .describe(
+      "solidIdentityProvider",
+      "The identity provider at which the user should authenticate."
+    )
+    .alias("idp", "solidIdentityProvider")
+    .describe("appName", "The name of the app you are registering.")
+    .describe(
+      "registration",
+      "[static] if you want to manually register the client, [dynamic] otherwise."
+    )
+    .describe(
+      "port",
+      "The port number on which the identity provider will return the token."
+    )
+    .locale("en")
+    .help().argv;
 
-const server = app.listen(PORT, async () => {
-  session = new Session({
+  const inputOptions: InputOptions = {
+    ...argv,
+    clientName: argv.appName,
+  };
+  // Complete CLI arguments with user prompt
+  const validatedOptions: ValidatedOptions = {
+    solidIdentityProvider:
+      inputOptions.solidIdentityProvider ?? (await promptIdp()),
+    registrationType:
+      inputOptions.registrationType ?? (await promptRegistration()),
+    clientName: inputOptions.clientName ?? (await promptClientName()),
+  };
+  const port = argv.port ?? (await promptPort());
+
+  const app = express();
+  const iriBase = `http://localhost:${port}`;
+  const storage = new InMemoryStorage();
+
+  const session: Session = new Session({
     insecureStorage: storage,
     secureStorage: storage,
   });
 
-  console.log(`Listening at: [http://localhost:${PORT}].`);
-  console.log(`Logging in ${argv.oidcIssuer} to get a refresh token.`);
-  session
-    .login({
-      clientName: argv.clientName,
-      oidcIssuer: argv.oidcIssuer,
+  const server = app.listen(port, async () => {
+    console.log(`Listening at: [${iriBase}].`);
+    const loginOptions: ILoginInputOptions = {
+      clientName: validatedOptions.clientName,
+      oidcIssuer: validatedOptions.solidIdentityProvider,
       redirectUrl: iriBase,
       tokenType: "DPoP",
       handleRedirect: (url) => {
         console.log(`\nPlease visit ${url} in a web browser.\n`);
       },
-    })
-    .catch((e) => {
+    };
+    let clientInfo;
+    if (validatedOptions.registrationType === "static") {
+      console.log(
+        `Please go perform the static registration of your application to [${validatedOptions.solidIdentityProvider}].`
+      );
+      console.log(`The redirect IRI will be ${iriBase}`);
```

Comment:
```suggestion
      console.log(`The redirect IRI will be [${iriBase}].`);
```
But why are we saying this? Do we require a 'redirect IRI' to perform that static registration? If so, then I'd say that here.

Reply:
Yes, the redirect IRI must be provided when doing static registration, which is indeed why it's mentioned. I'll rephrase it a bit.